tests/mqtt/test_adapter.py

```
@@ -0,0 +1,769 @@
+
+    @patch("paho.mqtt.client.Client.disconnect")
+    def test_error_is_logged_when_non_ok_return_code_is_returned_from_mqtt_client(self, mock: Mock):
+        mock.return_value = MQTT_ERR_SUCCESS + 1
+        self.adapter.connect()
+        with self.assertLogs(logger=_logger._logger, level=logging.ERROR):
+            self.adapter.disconnect()
+
+    def test_publishing_message_from_disconnected_client_logs_warning_and_raises_error(self):
+        msg = command("session_id", 0, Device(), b"some_command")
+        with self.assertLogs(logger=_logger._logger, level=logging.WARNING):
+            with self.assertRaises(MQTTCommunicationError):
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)